physical/oci_objectstorage/oci_objectstorage.go

```diff
@@ -0,0 +1,398 @@
+		opcClientRequestId = ""
+	}
+	o.logger.Debug("physical/oci: Delete", "opc-client-request-id", opcClientRequestId)
+	request := objectstorage.DeleteObjectRequest{
+		NamespaceName:      &o.namespaceName,
+		BucketName:         &o.bucketName,
+		ObjectName:         &key,
+		OpcClientRequestId: &opcClientRequestId,
+	}
+
+	resp, err := o.client.DeleteObject(ctx, request)
+	o.logRequest("delete", resp.RawResponse, resp.OpcClientRequestId, resp.OpcRequestId, err)
+
+	if err != nil {
+
+		if resp.RawResponse != nil && resp.RawResponse.StatusCode == http.StatusNotFound {
+			defer resp.RawResponse.Body.Close()
+			return nil
+		}
+
+		metrics.SetGauge(metricDeleteFailed, 1)
+		return errwrap.Wrapf("failed to delete Key: {{err}}", err)
+	}
+	o.logger.Debug("physical/oci: delete end")
+	defer resp.RawResponse.Body.Close()
+
+	return nil
+}
+
+func (o *Backend) List(ctx context.Context, prefix string) ([]string, error) {
+	o.logger.Debug("physical/oci: list")
+	defer metrics.MeasureSince(metricListFull, time.Now())
+	metrics.SetGauge(metricPermitsUsed, float32(o.permitPool.CurrentPermits()))
+	startAcquirePool := time.Now()
+	o.permitPool.Acquire()
+	defer o.permitPool.Release()
+
+	metrics.MeasureSince(metricListAcquirePool, startAcquirePool)
+	defer metrics.MeasureSince(metricList, time.Now())
+	var keys []string
+	delimiter := "/"
+	var start *string
+
+	for {
+
+		opcClientRequestId, err := uuid.GenerateUUID()
+		if err != nil {
+			o.logger.Debug("physical/oci: List: error generating UUID")
+			opcClientRequestId = ""
+		}
+		o.logger.Debug("physical/oci: List", "opc-client-request-id", opcClientRequestId)
+		request := objectstorage.ListObjectsRequest{
+			NamespaceName:      &o.namespaceName,
+			BucketName:         &o.bucketName,
+			Prefix:             &prefix,
+			Delimiter:          &delimiter,
+			Start:              start,
+			OpcClientRequestId: &opcClientRequestId,
+		}
+
+		response, err := o.client.ListObjects(ctx, request)
+		o.logRequest("list", response.RawResponse, response.OpcClientRequestId, response.OpcRequestId, err)
+
+		if err != nil {
+			metrics.SetGauge(metricListFailed, 1)
+
+			return nil, errwrap.Wrapf("failed to list using prefix: {{err}}", err)
+		}
+
+		for _, commonPrefix := range response.Prefixes {
+			commonPrefix := strings.TrimPrefix(commonPrefix, prefix)
+			keys = strutil.AppendIfMissing(keys, commonPrefix)
+		}
+
+		for _, object := range response.Objects {
+			key := strings.TrimPrefix(*object.Name, prefix)
+			keys = strutil.AppendIfMissing(keys, key)
+		}
+
+		if response.NextStartWith == nil {
+			response.RawResponse.Body.Close()
+			break
+		}
+
+		start = response.NextStartWith
+		response.RawResponse.Body.Close()
+	}
+
+	sort.Strings(keys)
+	o.logger.Debug("physical/oci: list end")
+	return keys, nil
+}
+
+func (o *Backend) logRequest(operation string, response *http.Response, clientOpcRequestIdPtr *string, opcRequestIdPtr *string, err error) {
+
+	statusCode := 0
```

Comment:
Response code 0 may be confusing. It might be better to default this to the string "no response", and then use `strconv.Itoa` in the block that sets it.